### PR TITLE
Upgrade to the 0.5 format

### PR DESCRIPTION
**Pull request by josmo**
Update to dockerfiles
Update to README
added .github files
govendor

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -2,127 +2,210 @@ package main
...
-
-	"github.com/drone/drone-go/drone"
-	"github.com/drone/drone-go/plugin"
-	"github.com/thoj/go-ircevent"
 )
 
-var (
-	buildCommit string
-)
+var build string
```
Comment (tboerger) [inline main.go]: Please add the comment about being set on build

File: main.go [hunk 2]
```diff
@@ -2,127 +2,210 @@ package main
...
+		cli.StringFlag{
+			Name:   "prefix",
+			Usage:  "prefix for notification",
+			EnvVar: "PLUGIN_PREFIX",
+			Value:  "build",
+		},
+		cli.StringFlag{
+			Name:   "nick",
+			Usage:  "nickname used by bot",
+			EnvVar: "PLUGIN_NICK,",
```
Comment (tboerger) [inline main.go]: Drop the comma

File: main.go [hunk 3]
```diff
@@ -2,127 +2,210 @@ package main
...
+		},
+		cli.StringFlag{
+			Name:   "channel",
+			Usage:  "channel to post message in",
+			EnvVar: "PLUGIN_CHANNEL,",
```
Comment (tboerger) [inline main.go]: Same here

File: main.go [hunk 4]
```diff
@@ -2,127 +2,210 @@ package main
...
+		},
+		cli.StringFlag{
+			Name:   "recipient",
+			Usage:  "recipient",
+			EnvVar: "PLUGIN_RECIPIENT",
+		},
+		cli.IntFlag{
+			Name:   "irc-host",
+			Usage:  "irc-host",
+			EnvVar: "PLUGIN_IRC_HOST",
```
Comment (tboerger) [inline main.go]: PLUGIN_HOST

File: main.go [hunk 5]
```diff
@@ -2,127 +2,210 @@ package main
...
+			Name:   "channel",
+			Usage:  "channel to post message in",
+			EnvVar: "PLUGIN_CHANNEL,",
+		},
+		cli.StringFlag{
+			Name:   "recipient",
+			Usage:  "recipient",
+			EnvVar: "PLUGIN_RECIPIENT",
+		},
+		cli.IntFlag{
```
Comment (tboerger) [inline main.go]: StringFlag

File: main.go [hunk 6]
```diff
@@ -2,127 +2,210 @@ package main
...
+			Name:   "irc-host",
+			Usage:  "irc-host",
+			EnvVar: "PLUGIN_IRC_HOST",
+			Value:  6667,
```
Comment (tboerger) [inline main.go]: This default should be moved to the port

File: main.go [hunk 7]
```diff
@@ -2,127 +2,210 @@ package main
...
+		},
+		cli.StringFlag{
```
Comment (tboerger) [inline main.go]: IntFlag

File: main.go [hunk 8]
```diff
@@ -2,127 +2,210 @@ package main
...
+			Name:   "irc-port",
+			Usage:  "irc-port",
+			EnvVar: "PLUGIN_IRC_PORT",
```
Comment (tboerger) [inline main.go]: PLUGIN_PORT

File: main.go [hunk 9]
```diff
@@ -2,127 +2,210 @@ package main
...
+		},
+		cli.StringFlag{
+			Name:   "irc-password",
+			Usage:  "irc-password",
+			EnvVar: "PLUGIN_IRC_PASSWORD",
```
Comment (tboerger) [inline main.go]: PLUGIN_PASSWORD,IRC_PASSWORD

File: main.go [hunk 10]
```diff
@@ -2,127 +2,210 @@ package main
...
+		},
+		cli.BoolFlag{
+			Name:   "irc-enable-tls",
+			Usage:  "enable-tls",
+			EnvVar: "PLUGIN_IRC_ENABLE_TLS",
```
Comment (tboerger) [inline main.go]: Also drop IRC_

File: main.go [hunk 11]
```diff
@@ -2,127 +2,210 @@ package main
...
+		},
+		cli.StringFlag{
```
Comment (tboerger) [inline main.go]: Some webhook parameters?

File: plugin.go [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+	"math/rand"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type (
+	Repo struct {
+		Owner string `json:"owner"`
```
Comment (tboerger) [inline plugin.go]: You can drop the json tags

File: plugin.go [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+		Repo   Repo
+		Build  Build
+		Config Config
+		Job    Job
+	}
+)
+
+func (p Plugin) Exec() error {
+
+	//system := drone.System{}
```
Comment (tboerger) [inline plugin.go]: You can drop this commented code

File: plugin.go [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+	//vargs := Params{}
+
+	//plugin.Param("system", &system)
+	//plugin.Param("repo", &repo)
+	//plugin.Param("build", &build)
+	//plugin.Param("vargs", &vargs)
+	//plugin.MustParse()
+
+	if len(p.Config.Channel) == 0 && len(p.Config.Recipient) == 0 {
+		fmt.Println("Please provide a channel or recipient")
```
Comment (tboerger) [inline plugin.go]: I think you can drop that, the error should be printed

File: plugin.go [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+			}
+		}
+
+		if strings.HasPrefix(destination, "#") {
+			client.Join(destination)
+		}
+
+		client.Privmsgf(
+			destination,
+			"[%s %s/%s#%s] %s on %s by %s (%s)",
```
Comment (tboerger) [inline plugin.go:122]: Maybe using a template for that?

**Review by tboerger — CHANGES_REQUESTED** [review]
Sorry for all the comments :(

**Comment by josmo** [comment]
thanks @tboerger and don't be sorry at all.  I need to get better at catching those!  Thanks so much for them!

File: plugin.go (hunk 15 above)
Comment (josmo) [reply plugin.go:122]: I wanted to do that, but I was thinking about putting that in a different PR

**Comment by josmo** [comment]
btw - right now everything works locally for notification but on my build server I get "use of closed network connection"  I'm trying to hunt down why that might be the case.

**Comment by josmo** [comment]
Ok the issue for the notifications not working on the server was that from aws freenode requires sasl. I added that and debuging.

File: Dockerfile [hunk 16]
```diff
@@ -1,15 +1,11 @@
...
-#
-#     cd $GOPATH/src/github.com/drone-plugins/drone-irc
-#     make deps build
-#     docker build --rm=true -t plugins/drone-irc .
+FROM plugins/base:multiarch
+MAINTAINER Drone.IO Community <drone-dev@googlegroups.com>
 
-FROM alpine:3.2
+LABEL org.label-schema.version=latest
+LABEL org.label-schema.vcs-url="https://github.com/drone-plugins/drone-ric.git"
```
Comment (tboerger) [inline Dockerfile]: Ric typo

File: main.go [hunk 17]
```diff
@@ -2,127 +2,218 @@ package main
...
+		},
+		cli.BoolFlag{
+			Name:   "use-sasl",
+			Usage: "use-sasl",
+			EnvVar: "PLUGIN_USE_SASL",
+		},
+		cli.BoolFlag{
+			Name: "debug",
+			Usage: "debug",
+			EnvVar:"PLUGIN_DEBUG",
```
Comment (tboerger) [inline main.go]: Missing space for proper formatting?

File: main.go [hunk 18]
```diff
@@ -2,127 +2,218 @@ package main
...
+		},
+		cli.StringFlag{
+			Name:   "template",
```
Comment (tboerger) [inline main.go]: This param have been posted from webhook plugin?

File: main.go [hunk 19]
```diff
@@ -2,127 +2,218 @@ package main
...
+			Usage:  "Custom template for webhook",
+			EnvVar: "PLUGIN_TEMPLATE",
+		},
+		cli.StringSliceFlag{
+			Name:   "headers",
```
Comment (tboerger) [inline main.go]: Same here

File: main.go [hunk 20]
```diff
@@ -2,127 +2,218 @@ package main
...
+			Usage:  "Custom headers key map",
+			EnvVar: "PLUGIN_HEADERS",
+		},
+		cli.StringSliceFlag{
+			Name:   "urls",
```
Comment (tboerger) [inline main.go]: Same here

File: plugin.go [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+
+	if client == nil {
+		return fmt.Errorf("Failed to create IRC Client: Invalid nick?")
+	}
+
+	client.Password = p.Config.IRCPassword
+	client.UseTLS = p.Config.IRCEnableTLS
+	client.Debug = p.Config.IRCDebug
+	client.UseSASL = p.Config.IRCSASL
+	if p.Config.IRCSASL{
```
Comment (tboerger) [inline plugin.go]: Missing space

**Review by tboerger — COMMENTED** [review]
Only nitpicking left

**Review by tboerger — APPROVED** [review]

**Comment by tboerger** [comment]
I need to create the webhook secret, than we can merge it

**Review by appleboy — APPROVED** [review]

**Comment by josmo** [comment]
@tboerger let me know when the secret is in and we can merge.  I have the update to use the normal template as an other PR when this is ready.

**Pull request merged**